docs/modules/ROOT/pages/reference/extensions/opentelemetry.adoc

```diff
@@ -0,0 +1,114 @@
+
+== What's inside
+
+* xref:{cq-camel-components}:others:opentelemetry.adoc[OpenTelemetry]
+
+Please refer to the above link for usage and configuration details.
+
+== Maven coordinates
+
+https://code.quarkus.io/?extension-search=camel-quarkus-opentelemetry[Create a new project with this extension on code.quarkus.io, window="_blank"]
+
+Or add the coordinates to your existing project:
+
+[source,xml]
+----
+<dependency>
+    <groupId>org.apache.camel.quarkus</groupId>
+    <artifactId>camel-quarkus-opentelemetry</artifactId>
+</dependency>
+----
+
+Check the xref:user-guide/index.adoc[User guide] for more information about writing Camel Quarkus applications.
+
+== Usage
+
+The extension automatically creates a Camel `OpenTelemetryTracer` and binds it to the Camel registry.
+
+In order to send the captured traces to a tracing system, you need to configure some properties within `application.properties` like those below.
+
+[source,properties]
+----
+# Identifier for the origin of spans created by the application
+quarkus.application.name=my-camel-application
+
+# For OTLP
+quarkus.opentelemetry.tracer.exporter.otlp.endpoint=http://localhost:55680
+
+# For Jaeger
+quarkus.opentelemetry.tracer.exporter.jaeger.endpoint=http://localhost:14268/api/traces
+----
+
+Note that you must add a dependency to the OpenTelemetry exporter that you want to work with. At present, Quarkus has support for
+Jaeger and the OpenTelemetry Protocol Specification (OTLP).
+
+For Jaeger:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-opentelemetry-exporter-jaeger</artifactId>
+</dependency>
+----
+
+For OTLP:
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-opentelemetry-exporter-otlp</artifactId>
+</dependency>
+----
+
+Refer to the https://github.com/quarkusio/quarkus/blob/{quarkus-version}/docs/src/main/asciidoc/opentelemetry.adoc[Quarkus OpenTelemetry guide] for a full list of configuration options.
+
+Route endpoints can be excluded from tracing by configuring a property named `quarkus.camel.opentelemetry.exclude-patterns` in `application.properties`. For example:
+
+[source,properties]
+----
+# Exclude all direct & netty-http endpoints from tracing
+quarkus.camel.opentelemetry.exclude-patterns=direct:*,netty-http:*
+----
+
+
+== Additional Camel Quarkus configuration
+
+[width="100%",cols="80,5,15",options="header"]
+|===
+| Configuration property | Type | Default
+
+
+|icon:lock[title=Fixed at build time] [[quarkus.camel.opentelemetry.encoding]]`link:#quarkus.camel.opentelemetry.encoding[quarkus.camel.opentelemetry.encoding]`
+
+Sets whether header keys need to be encoded
+| `boolean`
+| `false`
+
+|icon:lock[title=Fixed at build time] [[quarkus.camel.opentelemetry.exclude-patterns]]`link:#quarkus.camel.opentelemetry.exclude-patterns[quarkus.camel.opentelemetry.exclude-patterns]`
+
+Sets whether to disable tracing for endpoint URIs that match the given patterns
```

Comment:
Are they globs or regular expressions? I think as a reader I'd appreciate more clarity.

Reply:
... and if they are globs, like they seem to based on the example above, what's the syntax? I anything beyond `*` supported?

Reply:
Effectively these are the match rules:

https://github.com/apache/camel/blob/aa262669106bf21ce8d6b6c5e06608e300fa1ea2/core/camel-support/src/main/java/org/apache/camel/support/EndpointHelper.java#L205-L208

I can add some more notes to the config documentation.